internal/auth/credentials.go

```diff
@@ -0,0 +1,629 @@
+
+	storage := getTokenStorage(authMethod)
+
+	// Try keychain storage first
+	err := storage.SaveToken(token)
+	if err != nil {
+		// Fallback to file storage if keychain fails
+		if fileErr := saveTokenToFile(token, authMethod); fileErr != nil {
+			return fmt.Errorf("failed to save token to keychain (%w) and file storage (%w)", err, fileErr)
+		}
+		// Token saved to file successfully
+		return nil
+	}
+
+	return nil
+}
+
+// LoadTokenForMethod loads an OAuth2 token from the keychain using the specified authentication method key
+// Falls back to file storage if keychain operations fail or if --use-keychain=false
+func LoadTokenForMethod(authMethod string) (*oauth2.Token, error) {
+	// Check if user disabled keychain
+	if !shouldUseKeychain() {
+		// Directly load from file storage
+		return loadTokenFromFile(authMethod)
+	}
+
+	storage := getTokenStorage(authMethod)
+
+	// Try keychain storage first
+	token, err := storage.LoadToken()
+	if err != nil {
+		// Fallback to file storage if keychain fails
+		token, fileErr := loadTokenFromFile(authMethod)
+		if fileErr != nil {
+			return nil, fmt.Errorf("failed to load token from keychain (%w) and file storage (%w)", err, fileErr)
+		}
+
+		return token, nil
+	}
+
+	return token, nil
+}
+
+// SaveToken saves an OAuth2 token using device code storage for backward compatibility with older versions
+func SaveToken(token *oauth2.Token) error {
+	return SaveTokenForMethod(token, deviceCodeTokenKey)
+}
+
+// LoadToken attempts to load an OAuth2 token from the keychain, trying configured auth methods first,
+// then falling back to legacy storage for backward compatibility
+func LoadToken() (*oauth2.Token, error) {
+	// First, try to load using configuration-based keys from the active profile
+	authType, err := profiles.GetOptionValue(options.PingOneAuthenticationTypeOption)
+	if err == nil && authType != "" {
+		// Normalize "worker" to "client_credentials" for token loading
+		if authType == "worker" {
+			authType = "client_credentials"
+		}
+
+		// Try to get configuration for the configured auth method
+		var cfg *config.Configuration
+		var grantType svcOAuth2.GrantType
+		switch authType {
+		case "device_code":
+			cfg, _ = GetDeviceCodeConfiguration()
+			grantType = svcOAuth2.GrantTypeDeviceCode
+		case "auth_code":
+			cfg, _ = GetAuthCodeConfiguration()
+			grantType = svcOAuth2.GrantTypeAuthCode
+		case "client_credentials":
+			cfg, _ = GetClientCredentialsConfiguration()
+			grantType = svcOAuth2.GrantTypeClientCredentials
+		}
+
+		if cfg != nil {
+			// Set the grant type before generating the token key
+			cfg = cfg.WithGrantType(grantType)
+			tokenKey, err := GetAuthMethodKeyFromConfig(cfg)
+			if err == nil {
+				keychainStorage := svcOAuth2.NewKeychainStorage("pingcli", tokenKey)
+				token, err := keychainStorage.LoadToken()
+				if err == nil && token != nil {
+					return token, nil
+				}
+			}
+		}
+	}
+
+	// Also try all configured auth methods in case the type doesn't match
```

Comment:
This seems like unexpected behavior to me. If I specified client_credentials I would not want a token for device_code to be loaded from my keychain silently without warning.